expression/builtin_arithmetic.go

```diff
@@ -136,11 +141,13 @@ func (c *arithmeticDivideFunctionClass) setType4DivDecimal(retTp, a, b *types.Fi
 	}
 	if a.Flen == types.UnspecifiedLength {
 		retTp.Flen = mysql.MaxDecimalWidth
+		logutil.Logger(context.Background()).Warn("set field type Flen to MaxDecimalWidth", zap.Error(ErrUnknown.GenWithStackByArgs()))
```

Comment:
Please try to get the SQL statement from `ctx` by either passing `ctx` to `setType4DivDecimal`, or checking the `Flen` from the caller(https://github.com/pingcap/tidb/blob/b4644d433a8231de28c4e25bcc76c7c81e44c427/expression/builtin_arithmetic.go#L619)

Either way should be trivial enough.

Reply:
I add log in the caller function.